### DISCUSSION TITLE
Configurazione 2FA di Alexa Media Player

### DISCUSSION THREAD

**Discussion by Whitfy** [indomus/forum#1049] [HUB: Home Assistant]
Buongiorno a tutti,
ho una installazione di HA Core 2021.11.4 con Alexa Media Player 3.10.11. Ho qualche difficoltà ad aggiungere l'integrazione con 2FA. Di seguito riporto i passi che seguo:

1) Aggiungo integrazione e configuro la maschera come segue:
- Email + Password
- Imposto dominio "amazon.it"
- Imposto url al sito (https)
- Inserisco la chiave 2FA ricavata dal sito Amazon
- Lascio tutto il resto a valori dei default ("Login Proxy" YES, "enable oauth-token" YES) come da manaule del plugin

2) Nella schermata successiva mi viene fornito il codice OTP che inserisco sul sito Amazon
Clicco sulla checkbox "OTP from the Built-in 2FA App Key confirmed successfully" e proseguo

3) Si apre una schermata di login su Alexa dall'url del mio HA
- inserisco le credenziali e proseguo

4) La finestra mi chiede un codice OTP che è pre-configurato. In contemporanea ricevo un codice (diverso) anche sul telefono. Quello preimpostato non funziona, mentre quello su telefono si, ma se lo inserisco si riapre la pagina di login e tutto ricomincia.
Se invece seleziono "Codice OTP non ricevuto" posso scegliere "Inserisci OTP dall'app di autenticazione", ma in questo caso mi da un errore "Inserisci il tuo indirizzo e-mail o numero di cellulare, Inserisci la password".
[edit] Altre volte, inserendo il codice OTP fornito dal telefono, vengo indirizzato in una pagina "400: Bad Request"[/edit]

A questo punto sono in loop da cui non riesco a uscire e non riesco a installare l'integrazione :-(

Grazie a chiunque riesca ad aiutarmi su come uscirne!!

**Comment by Whitfy** [comment]
Giusto per informazione, ho provato a rifare l'integrazione seguendo gli stessi passaggi ma mettendo come dominio "amazon.com". In questo caso la configurazione termina con successo e importa tutti i miei dispositivi Alexa. Ovviamente poi non sono utilizzabili e sono tutti con stato "unknown".....

**Reply by fabiofi75** [reply]
Aggiornamenti? anche io sto impazzendo da 2 giorni e non riesco ad andare avanti.
quanto inserisco il codice OTP mi renindirizza nella pagina web ma mi da errore

**Reply by Whitfy** [reply]
Ciao, alla fine ho rimosso tutte le integrazioni di Alexa media Player (ne avevo una disattiva), disinstallato il plugin in HACS e rifatto tutta al procedura da zero. Così dopo un qualche tentativo ha funzionato.
